src/rust/ide/ast/impl/src/opr.rs

```diff
@@ -219,30 +276,117 @@ impl Chain {
     }
 
     /// Flattens infix chain if this is infix application of given operator.
     pub fn try_new_of(ast:&Ast, operator:&str) -> Option<Chain> {
-        let infix = GeneralizedInfix::try_new_root(&ast)?;
+        let infix = GeneralizedInfix::try_new(&ast)?;
         (infix.name() == operator).as_some_from(|| infix.flatten())
     }
 
     /// Iterates over &Located<Ast>, beginning with target (this argument) and then subsequent
     /// arguments.
-    pub fn enumerate_operands<'a>(&'a self) -> impl Iterator<Item=&'a Shifted<Located<Ast>>> + 'a {
-        let this = std::iter::once(&self.target);
-        let args = self.args.iter().map(|elem| &elem.operand);
-        this.chain(args).flatten()
+    pub fn enumerate_operands<'a>(&'a self) -> impl Iterator<Item=Located<&'a Shifted<Ast>>> + 'a {
+        let this_crumbs = self.args.iter().rev().map(ChainElement::crumb_to_previous).collect_vec();
+        let this        = self.target.as_ref().map(|opr| Located::new(this_crumbs,opr));
+        let args        = self.args.iter().enumerate().map(move |(i,elem)| elem.operand.as_ref().map(|opr| {
+            let to_infix   = self.args.iter().skip(i+1).rev().map(ChainElement::crumb_to_previous);
+            let has_target = self.target.is_some() || i > 0;
+            let crumbs     = to_infix.chain(elem.crumb_to_operand(has_target)).collect_vec();
+            Located::new(crumbs,opr)
+        }));
+        std::iter::once(this).chain(args).flatten()
     }
 
-    /// Iterates over &Located<Ast>, beginning with target (this argument) and then subsequent
-    /// arguments.
-    pub fn enumerate_operators<'a>(&'a self) -> impl Iterator<Item=&'a Located<known::Opr>> + 'a {
-        self.args.iter().map(|elem| &elem.operator)
+    /// Iterates over all operator's AST in this chain, starting from target side.
+    pub fn enumerate_operators<'a>(&'a self) -> impl Iterator<Item=Located<&'a known::Opr>> + 'a {
+        self.args.iter().enumerate().map(move |(i,elem)| {
+            let to_infix   = self.args.iter().skip(i+1).rev().map(ChainElement::crumb_to_previous);
+            let has_target = self.target.is_some() || i > 0;
+            let crumbs     = to_infix.chain(elem.crumb_to_operator(has_target)).collect_vec();
+            Located::new(crumbs,&elem.operator)
+        })
+    }
+
+    /// Insert new operand at index. The target's index is 0, the first argument index is 1, and so
+    /// on. So inserting at index 0 will actually set the new operand as a new target, and the old
+    /// target will became the first argument.
+    ///
+    /// Indexing does not skip `None` operands. Function panics, if get index greater than operands
+    /// count.
+    pub fn insert_operand(&mut self, at_index:usize, operand:Shifted<Ast>) {
+        let offset      = operand.off;
+        let mut operand = Some(operand);
+        let operator    = self.operator.clone_ref();
+        if at_index == 0 {
```

Comment:
Please extract `at_index == 0` to variable and describe semantically what it checks. Im trying to understand this code and Im failing to